larpandoracontent/LArHelpers/LArHitWidthHelper.cc

```diff
@@ -0,0 +1,278 @@
+/**
+ *  @file   larpandoracontent/LArHelpers/LArHitWidthHelper.cc
+ *
+ *  @brief  Implementation of the lar hit width helper class.
+ *
+ *  $Log: $
+ */
+#include "larpandoracontent/LArHelpers/LArHitWidthHelper.h"
+#include "larpandoracontent/LArHelpers/LArClusterHelper.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+LArHitWidthHelper::ConstituentHit::ConstituentHit(const CartesianVector &positionVector, const float hitWidth, const Cluster *const parentClusterAddress) :
+    m_positionVector(positionVector),
+    m_hitWidth(hitWidth),
+    m_parentClusterAddress(parentClusterAddress)
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+const Cluster* LArHitWidthHelper::ConstituentHit::GetParentClusterAddress() const
```

Comment:
There is a point in the HitWidthClusterMerging algorithm in which cluster parameters from two clusters are combined to compare their directions before and after a merge. At this point pParentClusterAddress is a nullptr. But this getter function is never actually used in the code, it is just there to make visualisation easier. So I will remove this check and make it an inline function

Reply:
sorry, that comment is really with respect to the ClusterParameters::GetClusterAddress().

Reply:
Hi - If you think there's any way (either of) these objects could dereference a nullptr, then please do keep the check. I can't immediately see how the pointers stored in these objects can be nullptr, unless that was what was provided at the point of constructing the instances.

In the case of clusters that have been merged, I can see that you can be left with a dangling pointer stored in the object, but it won't automatically be nullptr unless you've explicitly set it to be nullptr via your algorithm (or unless you use some kind of smart pointer).

What's your strategy for avoiding dereferencing dangling pointers? It's OK to have a container of e.g. deleted cluster pointers to you can compare addresses, so long as you definitely know what you're doing and don't do anything that tries to dereference the pointer and access the target memory (which would lead to undefined behaviour).

Hope that helps!